### PR TITLE
Chassis Bug Fix

### PR DESCRIPTION
**Pull request by 0-EricZhou-0**
Fixed a bug that causes the Update function to stuck when executing.

### CODE REVIEW

File: shared/libraries/chassis.cc [hunk 1]
```diff
@@ -65,20 +65,14 @@ Chassis::~Chassis() {
...
-      pids_ = nullptr;
-
       delete[] speeds_;
       speeds_ = nullptr;
       break;
   }
 }
 
 void Chassis::SetSpeed(const float x_speed, const float y_speed, const float turn_speed) {
+  print("Setspeed Start\r\n");
```
Comment (AzulRadio) [inline shared/libraries/chassis.cc:75]: Is this debug code?

File: shared/libraries/chassis.cc [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 namespace control {
 
-Chassis::Chassis(const chassis_t chassis) {
+Chassis::Chassis(const chassis_t chassis) : pids_() {
```
Comment (AzulRadio) [inline shared/libraries/chassis.cc:29]: What is this thing? Do you pass in a nullptr for pids_?

Comment (0-EricZhou-0) [reply shared/libraries/chassis.cc:29]: This is invoking the default constructor of `PIDController`, which sets Kp, Ki, Kd to all zeros. I reinit them afterwards.
It is formatted like this due to the constraint that `PIDController`s cannot be allocated using `new`.

File: shared/libraries/chassis.cc (hunk 1 above)
Comment (0-EricZhou-0) [reply shared/libraries/chassis.cc:75]: Good catch, I forgot to delete that.

**Review by AzulRadio — APPROVED** [review]

**Pull request merged**